x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/use_table_settings.ts

```diff
@@ -47,34 +38,29 @@ interface UseTableSettingsReturnValue<T extends object> {
 
 export function useTableSettings<TypeOfItem extends object>(
   sortByField: keyof TypeOfItem,
-  items: TypeOfItem[]
+  items: TypeOfItem[],
+  pageState: ListingPageUrlState,
+  updatePageState: (update: Partial<ListingPageUrlState>) => void
 ): UseTableSettingsReturnValue<TypeOfItem> {
-  const [tableSettings, setTableSettings] = useState<AnalyticsBasicTableSettings<TypeOfItem>>({
-    pageIndex: 0,
-    pageSize: PAGE_SIZE,
-    totalItemCount: 0,
-    showPerPageOptions: true,
-    sortField: sortByField,
-    sortDirection: 'asc',
-  });
+  const { pageIndex, pageSize, sortField, sortDirection } = pageState;
 
-  const onTableChange: EuiBasicTableProps<TypeOfItem>['onChange'] = ({
-    page = { index: 0, size: PAGE_SIZE },
-    sort = { field: sortByField, direction: 'asc' },
-  }: CriteriaWithPagination<TypeOfItem>) => {
-    const { index, size } = page;
-    const { field, direction } = sort;
+  const onTableChange: EuiBasicTableProps<TypeOfItem>['onChange'] = useCallback(
+    ({ page, sort }: CriteriaWithPagination<TypeOfItem>) => {
+      let resultSortField = sort?.field;
+      if (typeof resultSortField !== 'string') {
```

Comment:
what other types can it be?

Reply:
🤔 I think this is holdover defensive code from the example code I was using from https://github.com/elastic/kibana/pull/140613/files in `use_table_settings.ts` 
I think it depends on the dataset - it's possible it could not be a string type always.